oat-sa/tao-core
# Feature/ADF-1417/Upgrade to Node 18 (reopened)

### Related to: [ADF-1417](https://oat-sa.atlassian.net/browse/ADF-1417) and [ADF-1418](https://oat-sa.atlassian.net/browse/ADF-1418)

Requires:
 - [x] https://github.com/oat-sa/grunt-tao-bundle/pull/42
 - [x] https://github.com/oat-sa/tao-core-ui-fe/pull/558
 - [x] https://github.com/oat-sa/e2e-runner/pull/39
 - [x] update the file [`views/package.json`](https://github.com/oat-sa/tao-core/pull/3794/files#diff-03c6d048c4955944172f34af2a152ae6cbd4ffc2ec11fcd4bc7cd44b194f249cR16-R19) once the companion PRs have been merged and released
 - [x] update the file [`views/build/package.json`](https://github.com/oat-sa/tao-core/pull/3794/files#diff-a81a81309f0a9936d4d4db1b0cf0551c0be015e8793dc3e50295f4ae92bf1e01R22
) once the companion PRs have been merged and released


### Summary

Restored PR after the revert of the merge.

Update the dependencies in order to support further versions of Node

### Details

To properly support Node 18, the dev dependencies had to be updated and aligned.

BREAKING CHANGE:
- The package now requires Node.js 14.17+
- Replace Node-Sass with Dart-Sass

**Note:** The lock file was generated with `npm@6` so it would also work seamlessly with Node 14.

### How to test
- make sure to have Node 18 installed (you can use [nvm.sh](https://github.com/nvm-sh/nvm) for managing multiple versions)
- install the package:
    ```
    cd tao/views
    npm i
    cd build
    npm i
    npx grunt taosass
    npx grunt taobundle
    npx grunt connect:dev taotest
    ```

**Note** After switching to Node 18, you may need to erase the node_module folder, otherwise the error `>> Error: Could not locate the bindings file` may arise.
